### PR TITLE
bump golang 1.22.0 and update deps

### PR DESCRIPTION
**Pull request by astoycos**
Our prow based CI has been failing/flaking all over the place this seems to fix things. Update the controller-runtime and controller-tools dependencies, this requires some changes to the controller tools commands.

### CODE REVIEW

File: hack/update-codegen.sh [hunk 1]
```diff
@@ -48,12 +48,13 @@ export GOMODCACHE GO111MODULE GOFLAGS GOPATH
 # Even when modules are enabled, the code-generator tools always write to
 # a traditional GOPATH directory, so fake on up to point to the current
 # workspace.
-mkdir -p "$GOPATH/src/sigs.k8s.io"
-ln -sf "${SCRIPT_ROOT}" "$GOPATH/src/sigs.k8s.io/network-policy-api"
+# mkdir -p "$GOPATH/src/sigs.k8s.io"
```
Comment (astoycos) [inline hack/update-codegen.sh:51]: Ah nit I can delete these comments

File: hack/update-codegen.sh [hunk 2]
```diff
@@ -65,41 +66,43 @@ go run ./pkg/generator
...
+--output-file "zz_generated.register.go" \
 ${COMMON_FLAGS}
 
-echo "Generating ${VERSION} deepcopy at ${APIS_PKG}/apis/${VERSION}"
+echo "Generating ${API_VERSION} deepcopy at ${API_DIR}"
 go run sigs.k8s.io/controller-tools/cmd/controller-gen \
 object:headerFile="${SCRIPT_ROOT}/hack/boilerplate.generatego.txt" \
-paths="${APIS_PKG}/apis/${VERSION}"
\ No newline at end of file
+paths="${API_DIR}"
```
Comment (astoycos) [inline hack/update-codegen.sh:108]: Nit newline

File: go.mod [hunk 3]
```diff
@@ -1,33 +1,37 @@
 module sigs.k8s.io/network-policy-api
 
-go 1.21
+go 1.22.0
+
+toolchain go1.22.5
```
Comment (tssurya) [inline go.mod:5]: we shouldn't have this set

**Review by tssurya — CHANGES_REQUESTED** [review]

File: go.mod (hunk 3 above)
Comment (astoycos) [reply go.mod]: Done!

File: hack/update-codegen.sh [hunk 4]
```diff
@@ -45,15 +45,10 @@ fi
 
 export GOMODCACHE GO111MODULE GOFLAGS GOPATH
 
-# Even when modules are enabled, the code-generator tools always write to
-# a traditional GOPATH directory, so fake on up to point to the current
-# workspace.
-mkdir -p "$GOPATH/src/sigs.k8s.io"
-ln -sf "${SCRIPT_ROOT}" "$GOPATH/src/sigs.k8s.io/network-policy-api"
-
+readonly API_VERSION=v1alpha1
```
Comment (Dyanngg) [inline hack/update-codegen.sh:48]: I like that we now have a single place in the script for version. Let's maybe also add a reminder that everytime there's a spec bump this need to be updated

**Review by Dyanngg — APPROVED** [review]
LGTM overall

**Review by tssurya — APPROVED** [review]
/lgtm

**Pull request merged**